### ISSUE TITLE
Get rid of the interactive mode on cf cups -p

### ISSUE THREAD

**Issue by sjolicoeur** [cloudfoundry/cli#1201]
## Command  

cf cups -p "username, host, replaceme"

## What occurred

I was shown a prompt for the `username`, `host` and then `replaceme`

## What you expected to occur

For the command to just accept the parameters and continue. 

That said, I would just do away with the interactive prompt as it only add confusion. when [docs are written showing example usage it looks like the parameters should have a certain order and be swapped for their values](https://docs.cloudfoundry.org/devguide/services/user-provided.html).  Doing away with the interactive mode would make the docs clear, and make the command more unix-y. Also adding switches like:
 - `--host`
 - `--port`
 - `--password` ( this one could show an interactive prompt if no value is associated to it

and have them behave similarly to say the `mysql` command would be awesome. I think other commands using the concept of interactive prompt should also follow the same guideline.

## CLI Version  

cf version 6.29.0+ff886fa93.2017-07-24

## CC API Endpoint Version  

api version:    2.92.0

**Comment by dkoper** [comment]
Hi @sjolicoeur 

Not sure I understand the issue. You did `cf cups -p "username, host, replaceme"`, why and what was your expectation?

I have found the interactive behaviour unintuitive myself, but reluctant to remove functionality without deep consideration.
It would have been useful if it allowed you to enter credentials in a secure way, so it doesn't get captured in term history, but I see we always echo back what you type :(
Documentation can then explain that, addressing the confusion.

User provided services are treated in a generic way; we don't make assumptions on what parameters are relevant to the service or app.
Introducing `--host`, `--port` and `--password` flags, that are relevant to only a (big?) subset of services out there, yet probably incomplete for most, would add to confusion.

We could look at introducing a non-echoing prompt for parameters that are likely security sensitive parameters (e.g. "password", ".*token" - we can look at aligning to what we do globally to redact potentially sensitive values in CF trace) and make this feature more useful.

**Comment by sjolicoeur** [comment]
@dkoper I agree this decision cannot be taken lightly, but when reading the line:`cf cups -p "username, host, replaceme"` I expect to have to  enter `cf cups -p "some-username, host.something.com, something"`

That said, the reason I propose to remove it is that there is a perfectly flexible way to input data that has less ambiguity: the JSON file. at worse a son payload like `curl` allows for.

Regarding the security of echoing information or saving it to a file... I would argue that pretty much any User Provided service requires a password and making it an optional switch covers the  services that do not require one.

Another idea would be to allow the construction of the parameters in an additive manner like so: `-d key1=value1 -d key2=value2`

It's a hard position, which is why I value the removal of the interactive feature more so than adding switches that could add to the cognitive load.
